### PR TITLE
[hail] Convert ApplyAggOp and ApplyScanOp to use AggSignature2

### PR DESCRIPTION
**Pull request by tpoterba**
Sorry, you're getting the whole stack.

Stacked on #7658

### CODE REVIEW

**Comment by tpoterba** [comment]
passing!

**Comment by tpoterba** [comment]
bump

File: hail/src/main/scala/is/hail/expr/ir/agg/CollectAggregator.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ import is.hail.io.{BufferSpec, InputBuffer, OutputBuffer}
 
 class CollectAggregator(val elemType: PType) extends StagedAggregator {
 
-  val resultType = PArray(elemType, required = true)
+  val resultType = PArray(elemType, required = false)
```
Comment (catoverdrive) [inline hail/src/main/scala/is/hail/expr/ir/agg/CollectAggregator.scala]: Is this change just until we can handle requiredness correctly in the IR?

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/agg/CollectAggregator.scala]: yes.

**Pull request merged**